### PR TITLE
Projects: csi-driver-spiffe

### PR DESCRIPTION
**Pull request by JoshVanL**
This PR moves the documentation for the [csi-driver-spiffe](https://github.com/cert-manager/csi-driver-spiffe) project to the website docs.

Paired with https://github.com/cert-manager/csi-driver-spiffe/pull/15

We were also missing an entry for `csi-driver` in the projects `README.md`, so have added that here.

### CODE REVIEW

File: content/docs/projects/README.md [hunk 1]
```diff
@@ -15,6 +15,18 @@ These tools help with security, compliance and control.
 - [approver-policy](./approver-policy.md):
   a cert-manager **approver** that will automatically approve or deny
   certificate requests based on defined policy.
+- [csi-driver](./csi-driver.md):
+  csi-driver is a Container Storage Interface (CSI) driver plugin for Kubernetes
```
Comment (SgtCoDFish) [inline content/docs/projects/README.md:19]: nitpick (non-blocking): in other entries on this list we don't repeat the name of the project after the link. I think that stutters a bit; what do you think to removing it?

```suggestion
  a Container Storage Interface (CSI) driver plugin for Kubernetes
```

File: content/docs/projects/README.md [hunk 2]
```diff
@@ -15,6 +15,18 @@ These tools help with security, compliance and control.
...
+  to work along cert-manager. The goal for this plugin is to seamlessly request
+  and mount certificate key pairs to pods. This is useful for facilitating mTLS,
+  or otherwise securing connections of pods with guaranteed present certificates
+  whilst having all of the features that cert-manager provides.
+- [csi-driver-spiffe](./csi-driver-spiffe.md):
+  csi-driver-spiffe is another CSI driver plugin to work along
```
Comment (SgtCoDFish) [inline content/docs/projects/README.md:25]: nitpick (non-blocking): As above:

```suggestion
  a CSI driver plugin to work along
```

File: content/docs/projects/csi-driver-spiffe.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
+---
+title: csi-driver-spiffe
+description: ''
+---
```
Comment (SgtCoDFish) [inline content/docs/projects/csi-driver-spiffe.md:4]: note: Clearly copied directly from the csi-driver-spiffe README so I'm not going to dig into it too much :+1:

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve
/hold

Added two totally-not-required suggestions, so I'll add a hold. Happy to merge as-is, thanks for doing this!

File: content/docs/projects/README.md (hunk 1 above)
Comment (JoshVanL) [reply content/docs/projects/README.md:19]: Thanks, I agree. Let's remove the stutter

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve

Thanks :grin:

**Comment by JoshVanL** [comment]
flake

/retest

**Pull request merged**